<!-- This file demonstrates the use of SIFT features for online SLAM with a Kinect. 
     The openni driver is started from this file -->
<launch>
  <include file="$(find openni_launch)/launch/openni.launch"/>
  <node name="$(anon dynparam)" pkg="dynamic_reconfigure" type="dynparam" args="set_from_parameters /camera/driver">
    <param name="image_mode" type="int" value="5" /><!-- QVGA@30Hz -->
    <param name="depth_mode" type="int" value="5" /><!-- QVGA@30Hz -->
  </node>
  <node pkg="rgbdslam" type="rgbdslam" name="rgbdslam" cwd="node" required="true" output="screen"> 
    <!-- Input data settings-->
    <param name="config/topic_image_mono"              value="/camera/rgb/image_rect_color"/> <!--could also be color -->
    <param name="config/topic_image_depth"             value="/camera/depth_registered/image_rect"/>
    <param name="config/topic_points"                  value="/camera/depth_registered/points"/> <!--if empty, poincloud will be reconstructed from image and depth -->

    <param name="config/feature_detector_type"         value="SURF"/><!-- If SIFTGPU is enabled in CMakeLists.txt, use SURF here -->
    <param name="config/feature_extractor_type"        value="SURF"/><!-- If SIFTGPU is enabled in CMakeLists.txt, use SURF here -->
    <param name="config/nn_distance_ratio"             value="0.95"/> <!-- Feature correspondence is valid if distance to nearest neighbour is smaller than this parameter times the distance to the 2nd neighbour -->
    <param name="config/max_keypoints"                 value="300"/><!-- Extract no more than this many keypoints (not honored by SIFTGPU)-->
    <param name="config/visualization_skip_step"       value="1"/> <!-- draw only every nth pointcloud row and line, high values require higher squared_meshing_threshold -->
    <param name="config/observability_threshold"       value="-0.5"/>  <!--disabled-->
    <param name="config/depth_camera_fx"               value="295.0"/>
    <param name="config/depth_camera_fy"               value="295.0"/>
    <param name="config/depth_camera_cx"               value="160.0"/>
    <param name="config/depth_camera_cy"               value="120.0"/>
    <param name="config/min_translation_meter"         value="-1.0"/><!--disabled -->
    <param name="config/min_rotation_degree"           value="-1.0"/><!--disabled -->

  </node>
</launch>
